Simplified/NYPLMyBooksDownloadCenter.m

```diff
@@ -774,6 +799,36 @@ - (void)startDownloadForBook:(NYPLBook *const)book
   }
 }
 
+- (void)addDownloadTaskWithRequest:(NSURLRequest *)request
+                              book:(NYPLBook *)book {
+  NSURLSessionDownloadTask *const task = [self.session downloadTaskWithRequest:request];
+  
+  self.bookIdentifierToDownloadInfo[book.identifier] =
+    [[NYPLMyBooksDownloadInfo alloc]
+     initWithDownloadProgress:0.0
+     downloadTask:task
+     rightsManagement:NYPLMyBooksDownloadRightsManagementUnknown];
+  
+  self.taskIdentifierToBook[@(task.taskIdentifier)] = book;
+  
+  [task resume];
+  
+  [[NYPLBookRegistry sharedRegistry]
+   addBook:book
+   location:nil
+   state:NYPLBookStateDownloading
+   fulfillmentId:nil
+   readiumBookmarks:nil
+   genericBookmarks:nil];
+  
+  // It is important to issue this immediately because a previous download may have left the
+  // progress for the book at greater than 0.0 and we do not want that to be temporarily shown to
+  // the user. As such, calling |broadcastUpdate| is not appropriate due to the delay.
+  [[NSNotificationCenter defaultCenter]
+   postNotificationName:NYPLMyBooksDownloadCenterDidChangeNotification
```

Comment:
not sure I understand the reason to post this notification. At this point we just started a download, right? I thought this notification was for when the download has concluded....

Reply:
I believe it's to update the information shown to user if previous download failed. 
This part of code originally existed and I am only refactoring them into a function to avoid duplication.